.github/files/martin.markdownlint-cli2.jsonc

```diff
@@ -0,0 +1,18 @@
+{
```

Comment:
for consistency, lets rename this file as `.github/files/markdownlint-cli2.jsonc`

Reply:
I suspect we couldn't do this:
```shell
@sharkAndshark ➜ /workspaces/martin (doc_style_check) $ markdownlint-cli2 --config ./.github/files/markdownlint-cli2.jsonc --fix
markdownlint-cli2 v0.11.0 (markdownlint v0.32.1)
Error: Configuration file "/workspaces/martin/.github/files/markdownlint-cli2.jsonc" is unrecognized; its name should be (or end with) one of the supported types (e.g., ".markdownlint.json" or "example.markdownlint-cli2.jsonc").
    at readOptionsOrConfig (/usr/local/share/nvm/versions/node/v20.8.1/lib/node_modules/markdownlint-cli2/markdownlint-cli2.js:191:11)
    at main (/usr/local/share/nvm/versions/node/v20.8.1/lib/node_modules/markdownlint-cli2/markdownlint-cli2.js:935:13)
    at /usr/local/share/nvm/versions/node/v20.8.1/lib/node_modules/markdownlint-cli2/markdownlint-cli2.js:1045:32
    at run (/usr/local/share/nvm/versions/node/v20.8.1/lib/node_modules/markdownlint-cli2/markdownlint-cli2.js:1050:5)
    at Object.<anonymous> (/usr/local/share/nvm/versions/node/v20.8.1/lib/node_modules/markdownlint-cli2/markdownlint-cli2.js:1062:3)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
```
```js
// Read an options or config file in any format and return the object
const readOptionsOrConfig = async (configPath, fs, noRequire) => {
  const basename = pathPosix.basename(configPath);
  const dirname = pathPosix.dirname(configPath);
  let options = null;
  let config = null;
  if (basename.endsWith(".markdownlint-cli2.jsonc")) {
    const jsoncParse = await getJsoncParse();
    options = jsoncParse(await fs.promises.readFile(configPath, utf8));
  } else if (basename.endsWith(".markdownlint-cli2.yaml")) {
    options = yamlParse(await fs.promises.readFile(configPath, utf8));
  } else if (
    basename.endsWith(".markdownlint-cli2.cjs") ||
    basename.endsWith(".markdownlint-cli2.mjs")
  ) {
    options = await (
      importOrRequireConfig(fs, dirname, basename, noRequire, noop)()
    );
  } else if (
    basename.endsWith(".markdownlint.jsonc") ||
    basename.endsWith(".markdownlint.json") ||
    basename.endsWith(".markdownlint.yaml") ||
    basename.endsWith(".markdownlint.yml")
  ) {
    const jsoncParse = await getJsoncParse();
    config =
      await markdownlintReadConfig(configPath, [ jsoncParse, yamlParse ], fs);
  } else if (
    basename.endsWith(".markdownlint.cjs") ||
    basename.endsWith(".markdownlint.mjs")
  ) {
    config = await (
      importOrRequireConfig(fs, dirname, basename, noRequire, noop)()
    );
  } else {
    throw new Error(
      `Configuration file "${configPath}" is unrecognized; ` +
      "its name should be (or end with) one of the supported types " +
      "(e.g., \".markdownlint.json\" or \"example.markdownlint-cli2.jsonc\")."
    );
  }

  if (options) {
    if (options.config) {
      options.config = await getExtendedConfig(options.config, configPath, fs);
    }
    return options;
  }

  config = await getExtendedConfig(config, configPath, fs);
  return { config };
};
```